### DISCUSSION TITLE
Loading asset png-images does not work properly

### DISCUSSION THREAD

**Discussion by cgHome** [FranckFreiburger/vue3-sfc-loader#207] [Q&A]
Hi Frank,

I'm still relatively new to the Vue world and have encountered the following problem with the image loader. 

When I call the method “getContentData(true);” on an asset PNG image I get the following error message.

`warn unexpected data type. binary is expected for "../assets/vue/assets/eglass-filter.png"`

After a longer search (🤷‍♂️) I found the following solution. If the path contains a URL, then I call the method “getContentData(true)” otherwise I return the path variable.

Workaround:
```
               handleModule: async (type, getContentData, path, options) => {
                    switch (type) {
                        case ".png":
                            if (path.startsWith('http://') || path.startsWith('https://')) {
                                return getContentData(true);
                            } else {
                                return path;
                            }
                        case ".svg":
                            return path;
                    }
                },
```

Is this still a problem for you or is this the solution if you want to work with asset and “normal” images ?

Chris

**Comment by FranckFreiburger** [comment]
Hi Chris,
using the `handleModule` option is the way to go.
have a look at this example: https://github.com/FranckFreiburger/vue3-sfc-loader/blob/main/docs/examples.md#image-loading
